openvdb/openvdb/tree/LeafManager.h

```diff
@@ -763,8 +808,8 @@ class LeafManager
             for (typename LeafRange::Iterator it = range.begin(); it; ++it) op(*it, it.pos());
         }
         void join(const LeafReducer& other) { mLeafOp->join(*(other.mLeafOp)); }
-        LeafOp *mLeafOp;
-        const bool mOwnsOp;
+        std::unique_ptr<LeafOp> mLeafOpPtr;
+        LeafOp *mLeafOp = nullptr;
```

Comment:
Is extra complexity worth the `LeafReducer` simply copying the op in the main constructor?

Reply:
It entirely depends on how expensive a deep-copy of the op is and how many threads you're running the reduction on? There's no stipulation that an op should _not_ store large amounts of data inside it. I mean we would probably advise against it, but this pattern at least mitigates against the worst outcome.